app/Mails/MailJet/MailjetMailer.php

```diff
@@ -17,15 +17,15 @@
     public function __construct(
         private Client $client,
         private LoggerInterface $logger,
-        private bool $sandboxMode
+        private SandboxHelper $sandboxHelper,
     ) {
 
     }
 
     public function send(Address $from, Addresses $to, int $templateId, string $subject, array $variables = []): void
     {
         $body = [
-            'SandboxMode' => $this->sandboxMode,
+            'SandboxMode' => $this->sandboxHelper->getSandboxMode($to),
```

Comment:
I think there is a potential risk here! The `$to` is a list of email addresses, when one of them is an allowed domain, all the other addresses also receive an email.

Reply:
Currently we always sent a mail one at at time because we sent as variables the firstname/lastname of each contact.
So for now this is not a problem.
So this would only be a problem for future changes, I think we have 3 options here)

1) Keep it as is but put a comment on the class/function warning for this flaw.
2) Block all emails
3) More difficult - Manipulate the Addresses to change the collection to only contain valid emails. Not a fan of this because of the complexity, we don't need this currently.

What would you prefer?

Reply:
I see a fourth option: only disable the sandbox mode when all email addresses match the allowed domains. Putting a comment will be to tricky, this could result in sending mails to real integrators.

Reply:
Went for option 5